src/constructs/ec2/vpc.ts

```diff
@@ -1,17 +1,33 @@
 import type { ISubnet, IVpc, VpcAttributes } from "@aws-cdk/aws-ec2";
 import { Subnet, Vpc } from "@aws-cdk/aws-ec2";
 import type { GuStack } from "../core";
+import { GuVpcParameter } from "../core";
+
+interface VpcFromIdProps extends Omit<VpcAttributes, "availabilityZones"> {
+  availabilityZones?: string[];
+}
+
+type VpcFromIdParameterProps = Omit<VpcFromIdProps, "vpcId">;
 
 export class GuVpc {
   static subnets(scope: GuStack, subnets: string[]): ISubnet[] {
     return subnets.map((subnetId) => Subnet.fromSubnetId(scope, `subnet-${subnetId}`, subnetId));
   }
 
-  static fromId(scope: GuStack, id: string, vpcId: string, props?: Partial<VpcAttributes>): IVpc {
+  static fromId(scope: GuStack, id: string, props: VpcFromIdProps): IVpc {
     return Vpc.fromVpcAttributes(scope, id, {
-      vpcId: vpcId,
       availabilityZones: scope.availabilityZones,
       ...props,
     });
   }
+
+  static fromIdParameter(scope: GuStack, id: string, props?: VpcFromIdParameterProps): IVpc {
+    const vpc = new GuVpcParameter(scope, "VpcId", {
+      description: "Virtual Private Cloud to run EC2 instances within",
+      default: "/account/services/default.vpc",
```

Comment:
this might be a good opportunity to discourage use of the default VPC?

Reply:
Ah, good idea! We're intending the value of this param to be a non-default VPC - maybe a better name is `/account/services/primary.vpc`?

Reply:
I'm guilty of using default here to indicate the main VPC rather than the default in the AWS sense. `primary` is much better.

I would argue that VPCs warrant their own sub key. How about something like:
```
/account/vpc/primary/id
/account/vpc/primary/subnets/public
/account/vpc/primary/subnets/private
/account/vpc/primary/egress/ips
...
```

Reply:
Ah yes, that's nice. Then we can make add a similar method around subnets too.